### PR TITLE
feat: localstore container cache interfaces

### PR DESCRIPTION
**Pull request by aloknerurkar**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Cache-related interfaces in the localstore container. The difference between Cache and other components is that the cache is instantiated once and its size can be configured. Each Get and Put operation needs to be guarded by a transaction as these may include a few cache-related updates.

Also added tests to cover some error cases.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3755)

### CODE REVIEW

File: pkg/localstorev2/internal/cache/cache.go [hunk 1]
```diff
@@ -264,12 +264,12 @@ func (c *Cache) pushBack(
 
 // Putter returns a Storage.Putter instance which adds the chunk to the underlying
 // chunkstore and also adds a Cache entry for the chunk.
-func (c *Cache) Putter(store storage.Store, chStore storage.ChunkStore) storage.Putter {
+func (c *Cache) Putter(storg internal.Storage) storage.Putter {
```
Comment (mrekucci) [inline pkg/localstorev2/internal/cache/cache.go:267]: `storg` -> `store` or https://www.abbreviations.com/serp.php?st=Storage&category=IT&qtype=2
The same is below.

File: pkg/localstorev2/localstore.go [hunk 2]
```diff
@@ -348,3 +389,34 @@ func (db *DB) Pins() ([]swarm.Address, error) {
 func (db *DB) HasPin(root swarm.Address) (bool, error) {
 	return pinstore.HasPin(db.repo.IndexStore(), root)
 }
+
+// Lookup is the implementation of the CacheStore.Lookup function.
```
Comment (mrekucci) [inline pkg/localstorev2/localstore.go:393]: Change `function` to `method`. The same is above and below.

File: pkg/storagev2/repository.go [hunk 3]
```diff
@@ -12,31 +12,38 @@ import (
 
 // Repository is a collection of stores that provides a unified interface
 // to access them. Access to all stores can be guarded by a transaction.
-type Repository struct {
+type Repository interface {
```
Comment (mrekucci) [inline pkg/storagev2/repository.go:15]: What is the purpose of having a `Repository` interface?

Comment (aloknerurkar) [reply pkg/storagev2/repository.go:15]: So in the localstore I felt like we should have some tests related to errors. In order to achieve that I had to override the repository in the DB pointer. As the repository is a pointer it is not possible to extend it and change behavior.

Check the new tests in localstore_test.go related to rollbacks.

**Review by mrekucci — APPROVED** [review]

**Review by notanatol — APPROVED** [review]

**Pull request merged**